### ISSUE TITLE
Basic DataVec for Sequences Lab? 

### ISSUE THREAD

**Issue by tomthetrainer** [tomthetrainer/June26Class#5]
@turambar @bpark738 

I like to start with an early Lab, I will have them do simplest network that is basically the simplest NN you could imagine, that Lab is really just to verify they have a working environment and can use IntelliJ. 

The second Lab I have is a DataVec Lab. 

I demo a spark analysis from the demos section
AbaloneDataTransform 

For this class it might be nice to demo something more appropriate to time series. 

After the demo I have a basic DataVec Lab, 

The code is DataVecLab, basic ingest of iris csv data with a prebuilt Neural Network. 

It might be nice to have a basic Sequence oriented DataVec Lab. 

I could write it but if either of you wanted to summarize the issues with reading time series data and the datavec classes involved that might make a nice intro.

**Comment by turambar** [comment]
@tomthetrainer perhaps skim our physionet examples and see if the datavec stuff there suffices?

Briefly, I think the main challenges are:
- you often can't do one-record-per-row in a CSV, at least for multivariate time series
  - it's more common to do one-record-per-file, with one time step/observation per row
  - but that creates some challenges (Spark, for example, doesn't deal with with lots of individual files -- maybe @bpark738 can comment on that)
- for classification, you typically have variable length sequences but only a single target (label). Conceptually, that's simple to handle, but in practice, you have to decide how to align your label/target to your sequence. Simplest approach is to align the target with the last step and assume the other steps have no target. But you can also, e.g., replicate your target at each step.
- for forecasting, you have to decide whether and how to preprocess your data: do you turn a single sequence into multiple separate examples using a sliding window, or do you just train on the full sequence using TBPTT.

Any other comments @bpark738?

**Comment by tomthetrainer** [comment]
@turambar do you have content that describes these issues? Maybe slides from your GTC presentation? I would like to summarize what you just said into some content for the class

**Comment by turambar** [comment]
Alas no, but I can generate some over next day or so.

**Comment by bpark738** [comment]
Yeah not sure how to deal with individual files for Spark. I'll message Alex Black and see if he recommends anything.

The only other challenge I can think of is the training time required for LSTM's, especially if there are a high number of time steps. This might make using Spark more critical.
